src/clp_logging/readers.py

```diff
@@ -354,35 +356,49 @@ class CLPStreamReader(CLPBaseReader):
     """
 
     def __init__(
-        self, stream: IO[bytes], timestamp_format: Optional[str] = None, chunk_size: int = 4096
+        self,
+        stream: IO[bytes],
+        timestamp_format: Optional[str] = None,
+        chunk_size: int = 4096,
+        enable_compression: bool = True,
     ) -> None:
         super().__init__(timestamp_format, chunk_size)
         self.stream: IO[bytes] = stream
-        self.dctx: ZstdDecompressor = ZstdDecompressor()
-        self.zstream: ZstdDecompressionReader = self.dctx.stream_reader(
-            self.stream, read_across_frames=True
-        )
+        self.dctx: Optional[ZstdDecompressor] = None
+        self.zstream: Optional[ZstdDecompressionReader] = None
+        if enable_compression:
+            self.dctx = ZstdDecompressor()
+            self.zstream = self.dctx.stream_reader(self.stream, read_across_frames=True)
 
     def readinto_buf(self, offset: int) -> int:
         """
-        Use Zstandard to decompress the CLP IR stream.
+        Read the CLP IR stream directly or through Zstandard decompression.
         :return: Bytes read (0 for EOF), < 0 on error
         """
-        return self.zstream.readinto(self.view[offset:])
+        if self.zstream:
+            return self.zstream.readinto(self.view[offset:])
+        else:
+            # see https://github.com/python/typing/issues/659
+            return self.stream.readinto(self.view[offset:])  # type: ignore
 
     def close(self) -> None:
-        self.zstream.close()
+        if self.zstream:
+            self.zstream.close()  # type: ignore
```

Comment:
`self.zstream` has `closefd=True` by default, so the next line is not necessary.

Reply:
By not necessary, I mean it's not necessary in the compressed case.